session/src/main/java/org/apache/iotdb/session/SessionConnection.java

```diff
@@ -722,20 +748,28 @@ protected void testInsertTablets(TSInsertTabletsReq request)
   private boolean reconnect() {
     boolean flag = false;
     for (int i = 1; i <= Config.RETRY_NUM; i++) {
-      try {
-        if (transport != null) {
-          close();
-          init(endPoint);
-          flag = true;
-        }
-      } catch (Exception e) {
-        try {
-          Thread.sleep(Config.RETRY_INTERVAL_MS);
-        } catch (InterruptedException e1) {
-          logger.error("reconnect is interrupted.", e1);
-          Thread.currentThread().interrupt();
+      if (transport != null) {
+        transport.close();
+        int currHostIndex = endPointList.indexOf(session.defaultEndPoint);
+        for (int j = currHostIndex; j < endPointList.size(); j++) {
```

Comment:
not right, can not try the previous endpoint.

Reply:
If `currHostIndex` is the last index of `endPointList`, in your implementation, it can not try the others nodes again.

Reply:
If all the endpoints in the list fail, I will try to connect again with the head node, traversing the list RETRY_NUM times in total